website/server/libs/user/index.js

```diff
@@ -27,6 +27,7 @@ export async function get (req, res, { isV3 = false }) {
     userToJSON.needsCron = false;
     if (daysMissed > 0) userToJSON.needsCron = true;
     User.addComputedStatsToJSONObj(userToJSON.stats, userToJSON);
+    userToJSON.flags.newStuff = await user.checkNewStuff();
```

Comment:
so `newStuff` has been removed from the user model but we still want to return it to the user? Only in this route?

Reply:
We still want it on the user for backwards compatability. Since clients are checking for that field to determine if the user has unread news posts.

Reply:
makes sense! In this case I'd keep a commented out version of the old field in the schema to remember where it's coming from. Like `// newStuff: { $type: Boolean, default: false },` with a small explanation. 

Also this way it'll only be shown from this route which could be a problem, at the same time it's an async function and we don't want to really call it every time the user is returned

Reply:
Added that.
Technically this is an async function, but it should really only run async code for the first user after the server is started. After that the date of the last post is cached and retrieved without running any async code. Not sure if there is a better way to put this call, so that it is always returned when appropriate.

Reply:
Hmm yeah the problem is that even if it's doesn't make any asyn request it being a promise causes it to be handled as one, and there's the problem that this way only this route will have the updated value, not all the others where the user is returned. I think we'll have to find a different solution here

Reply:
So I think this should be moved to `toJSONTransform` here https://github.com/HabitRPG/habitica/blob/bd8e67a2ea3d6109ab2e43057ecdfe8f1be2e363/website/server/models/user/hooks.js#L25 so that it's present in every request and not only when getting the user. Also to avoid the issue with the `await`, if we refresh the last cached post with a callback called at an interval of X minutes (using `setInterval`) and not with a call to `await NewsPost.lastNewsPostID();` we can then skip the async call